ui/panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts

```diff
@@ -196,3 +195,27 @@ export function getThresholdSeries(
     },
   };
 }
+
+/**
+ * Converts percent threshold into absolute step value
+ * If max is undefined, use the max value from time series data as default
+ */
+export function convertPercentThreshold(percent: number, data: EChartsTimeSeries[], max?: number, min?: number) {
```

Comment:
Does this get used by GaugeChart too, or just TimeSeriesChart (I remembering seeing percentage mode for GaugeChart, so wondering what's doing that calculation for GaugeChart)

Reply:
Each chart has its own convert function because this is basically converting threshold values into something a basic chart can understand and be able to render it. For example, we convert threshold values into EChartsAxisLineColors for gauge chart.